galeracluster/source/faq.rst

```diff
@@ -5,24 +5,113 @@
+Galera Cluster is a write-set replication service provider in the form of the dlopenable library.  It provides synchronous replication and supports multi-master replication.  Galera Cluster is capable of unconstrained parallel applying (that is, "parallel replication"), multicast replication and automatic node provisioning.
 
-Occasionally, a slave thread tries to apply a replicated write-set and finds a lock conflict with a local transaction, which may already be in the commit phase. The local transaction is aborted and the slave thread can proceed. This is a consequence of an optimistic (expecting no row conflicts) transaction execution, and is expected in a multi-master configuration. You may see any of the following messages:
+The primary focus of Galera Cluster is data consistency.  Transactions are either applied to every node or not at all.  Galera Cluster is not a cluster manager, a load balancer or a cluster monitor.  What it does it keep databases synchronized provided that they were properly configured and synchronized in the beginning.
 
-.. code-block:: text
+.. rubric:: What is Galera?
+.. _`what-is-galera`:
+
+The word *galera* is the Italian word for *galley*.  The galley is a class of naval vessel used in the Mediterranean Sea from the 2nd millennium :sub:`B.C.E.` until the Renaissance.  Although they used sails when the winds were favorable, their principal method of propulsion came from banks of oars.
+
+In order to manage the vessel effectively, rowers had to act synchronously, lest the oars become intertwined and get blocked.  Captains could scale the crew up to hundreds of rowers, making the galleys faster and more maneuverable in combat.
+
+.. seealso:: For more information on galleys, see `Wikipedia <http://en.wikipedia.org/wiki/Galley>`_.
+
+
+
+.. rubric:: How Do I Manage Failover?
+.. _`failover`:
+
+Galera Cluster is a true synchronous multi-master replication system, which allows the use of any or all of the nodes as master at any time without any extra provisioning.  What this means is that there is no failover in the traditional MySQL master-slave sense.
+
+The primary focus of Galera Cluster is data consistency across the nodes.  This does not allow for any modifications to the database that may compromise consistency.  For instance, the node blocks or rejects write requests until the joining node syncs with the cluster and is ready to process load requests.
+
+The results of this is that you can safely use your favorite approach to distribute or migrate connections between the nodes without the risk of causing inconsistency.
+
+.. seealso:: For more information on connection distribution, see :doc:`deploymentvariants`.
+
+
+
+.. rubric:: How Do I Upgrade the Cluster?
+.. _`faq-upgrade`:
+
+Periodically, updates will become available for Galera Cluster.  The database server itself or the :term:`Galera Replication Plugin`.  To update the software for the node, complete the following steps:

```

Comment:
"The database server itself or the :term:`Galera Replication Plugin`." is a sentence fragment.